src/routing_message_filter.rs

```diff
@@ -70,9 +72,15 @@ impl RoutingMessageFilter {
 
     // Filter outgoing `RoutingMessage`. Return whether this specific message has been seen recently
     // (and thus should not be sent, due to deduplication).
+    //
+    // Return `false` if serialisation of the message fails - that can be handled elsewhere.
```

Comment:
Well spotted Michael. Qi may be right about returning `Result` but that distinction doesn't seem especially important.

Perhaps it would be sufficient to add a `debug`/`trace`/`warn` message when serialisation fails?